feature-libs/customer-ticketing/components/customer-ticketing-details.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import {
+  CustomerTicketingFacade,
+  STATUS,
+  Status,
+  TicketDetails,
+  TicketEvent,
+} from '@spartacus/customer-ticketing/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+} from '@spartacus/core';
+import { LaunchDialogService } from '@spartacus/storefront';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CustomerTicketingDetailsService {
+  dataLoading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  protected ticketDetails$ = this.customerTicketingService.getTicket();
+
+  constructor(
+    protected customerTicketingService: CustomerTicketingFacade,
+    protected globalMessageService: GlobalMessageService,
+    protected launchDialogService: LaunchDialogService,
+    protected routingService: RoutingService
+  ) {}
+
+  getTicketSubject = (): Observable<string | undefined> =>
+    this.ticketDetails$.pipe(map((details) => details?.subject));
+
+  getTicketStatus = (): Observable<string | undefined> =>
+    this.ticketDetails$.pipe(
+      map((details) => details?.status?.id?.toUpperCase())
+    );
+
+  getAvailableTransitionStatus = (): Observable<Status[] | undefined> =>
+    this.ticketDetails$.pipe(
+      map((details) => details?.availableStatusTransitions)
+    );
+
+  getTicketDetails = (): Observable<TicketDetails | undefined> =>
```

Comment:
you don't need all these functions as you derive these from the getTicketDetails() { .. } function instead

I believe it's just over complicating the stream as you are combining them in the end in the components.

The original point for this 'component' details service was to
1 - get the specific customer ticket through the current route (where the customer id is at)

Reply:
let's get in a call for this on monday